src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -65,20 +65,12 @@ Shard::open(Section config, Scheduler& scheduler, nudb::context& ctx)
         return false;
     }
 
-    boost::system::error_code ec;
-    auto const preexist {exists(dir_, ec)};
-    if (ec)
-    {
-        JLOG(j_.error()) <<
-            "shard " << index_ << ": " << ec.message();
-        return false;
-    }
-
     config.set("path", dir_.string());
     backend_ = factory->createInstance(
         NodeObject::keyBytes, config, scheduler, ctx, j_);
 
-    auto fail = [&](std::string msg)
+    auto const preexist {exists(dir_)};
```

Comment:
Until we are officially on C++17, combining list initialization with `auto` is dangerous.  For me (using Clang) `preexist` is being deduced as `const bool`, I'm not sure why.  But I think this declaration is supposed to yield `preexist` as `const std::initializer_list<bool>` in C++14.  See, e.g., https://mariusbancila.ro/blog/2017/04/13/cpp17-new-rules-for-auto-deduction-from-braced-init-list/.

So my habit has been to always specify the type when I use list initialization.  And that gives me an excuse to not use `auto` :wink:.  My preferred form for this line would be:
```
bool const preexist {exists(dir_)};
```
1. You know that you get a `const bool` and can read it in the source file.
2. Because you're using list initialization without an equals you know there is no narrowing.

Reply:
@scottschurr Thanks.